src/middlewared/middlewared/plugins/pool_/import_pool.py

```diff
@@ -204,117 +204,166 @@ async def import_pool(self, job, data):
 
         return True
 
+    @private
+    def import_on_boot_impl(self, vol_name, vol_guid, set_cachefile=False):
+        cmd = [
+            'zpool', 'import',
+            vol_guid,  # the GUID of the zpool
+            '-R', '/mnt',  # altroot
+            '-m',  # import pool with missing log device(s)
+            '-o', f'cachefile={ZPOOL_CACHE_FILE}' if set_cachefile else 'cachefile=none',
+        ]
+        try:
+            self.logger.debug('Importing %r with guid: %r', vol_name, vol_guid)
+            cp = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            if cp.returncode != 0:
+                self.logger.error(
+                    'Failed to import %r with guid: %r with error: %r',
+                    vol_name, vol_guid, cp.stdout.decode()
+                )
+                return False
+        except Exception:
+            self.logger.error('Unhandled exception importing %r', vol_name, exc_info=True)
```

Comment:
I wonder if we should generate a one-shot alert if pool fails to import.

Reply:
We already have "a pool is offline" alert and it'll be re-checked at boot time a few moments later.